classes/class-uninstall.php

```diff
@@ -81,9 +68,10 @@ public function uninstall() {
 
 		/**
 		 * Drop everything on single site installs or when network activated
-		 * Otherwise only delete data relative to the current blog.
+		 * AND being uninstalled from the main blog. Otherwise only delete
+		 * data relative to the current blog.
 		 */
-		if ( ! is_multisite() || $this->plugin->is_network_activated() ) {
+		if ( ! is_multisite() || ( $this->plugin->is_network_activated() && is_main_site() && is_super_admin() ) ) {
```

Comment:
@schlessera This fixes a really important issue where the plugin being uninstalled from a blog on a multisite would delete ALL Stream data, if the Stream plugin was activated at the network level.

We now purge all plugin records and tables only:

1. if the uninstall is happening on a single site, or
2. if the uninstall is happening on the main site of the multisite and is triggered by the super admin.

Reply:
Hmm, doesn't that mean that it won't clean up anymore if you only installed it on a subsite?

Reply:
The Stream tables are shared between all sites on a multisite.

Currently the tables are completely removed only if:

1. The plugin is "Uninstalled" from the main blog of the multisite _and_ it was enabled at the network level.
2. The plugin is "Uninstalled" from a non-multisite setup.

All other uninstalls simply delete the data associated with the particular `blog_id`. So in case of uninstalling the plugin from a blog on a network (when the plugin is not activated globally) it would preserve the tables and remove the data. Uninstalling it from the main blog will delete the tables only if the plugin was installed globally.

This changeset is attempting to prevent full table removal when a plugin is uninstalled from any single blog on the network when the plugin is activated at the network level.